app/controllers/collection_numbers_controller.rb

```diff
@@ -350,7 +350,7 @@ def figure_out_where_to_go_back_to
       @back_object = Observation.safe_find(@back)
       return if @back_object
 
-      @back_object = if @collection_number.observations.count == 1
+      @back_object = if @collection_number.observations.length == 1
```

Comment:
Another place for `one?`, which is defined as
```ruby
# File activerecord/lib/active_record/relation.rb, line 282
def one?
  return super if block_given?
  limit_value ? records.one? : size == 1
end
```
https://github.com/rails/rails/blob/56bcc0abd3c9a6b09469e9428f6eea0dd77c2294/activerecord/lib/active_record/relation.rb#L282

Note: I'm fine with not using `one?` here and elsewhere. I'm just putting it out as an option.

Reply:
Oops. Probably no on `size`, stick with `length`, as `observations` is used on the next line.

Reply:
OK. I went with `one?` - If you get a chance please have another look over everything — i've changed everything in the PR — and check if it seems right.